lightning-background-processor/src/lib.rs

```diff
@@ -673,9 +686,32 @@ where
 				task::Poll::Ready(exit) => { should_break = exit; true },
 				task::Poll::Pending => false,
 			}
-		}, mobile_interruptable_platform)
+		}, mobile_interruptable_platform
+	)
+}
+
+#[cfg(feature = "futures")]
+async fn process_onion_message_handler_events_async<
+	EventHandlerFuture: core::future::Future<Output = ()>,
+	EventHandler: Fn(Event) -> EventHandlerFuture,
+	PM: 'static + Deref + Send + Sync,
+>(
+	peer_manager: &PM, handler: EventHandler
+)
+where
+	PM::Target: APeerManager + Send + Sync,
+{
+	use lightning::events::EventsProvider;
+
+	let events = core::cell::RefCell::new(Vec::new());
+	peer_manager.onion_message_handler().process_pending_events(&|e| events.borrow_mut().push(e));
```

Comment:
Maybe after async functions in traits are stabilized, we can add an async event processing method to `EventsProvider` with a default implementation 🤔 I'm just wondering if this could be a temporary workaround.

Reply:
Yeah, possibly. `EventsProvider` is kinda weird in that up until now it never really needed to be a trait.